fbw-common/src/wasm/systems/systems/src/hydraulic/command_sensor_unit.rs

```diff
@@ -0,0 +1,336 @@
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum CSU {
+    Conf0 = 0,
+    Conf1,
+    Conf2,
+    Conf3,
+    ConfFull,
+    OutOfDetent,
+    Fault,
+    Misadjust, // This is triggered when flaps lever moves from 0 to 4 or viceversa
+               // with no intermediate steps. Not implemented because it can impact the simulation.
+}
+impl CSU {
+    pub fn is_valid(value: CSU) -> bool {
+        matches!(
+            value,
+            CSU::Conf0 | CSU::Conf1 | CSU::Conf2 | CSU::Conf3 | CSU::ConfFull
+        )
+    }
+
+    // The CSU is made of two parallel switches: this is why there is `switch_pattern_1` and
+    // `switch_pattern_2`.
+    pub fn get_csu_configuration(
+        switch_pattern_1: SwitchPattern,
+        switch_pattern_2: SwitchPattern,
+    ) -> CSU {
+        match (switch_pattern_1, switch_pattern_2) {
+            (0b11000, 0b11000) => CSU::Conf0,
+            (0b11000, 0b01000) => CSU::OutOfDetent,
+            (0b01000, 0b01000) => CSU::OutOfDetent,
+            (0b01000, 0b01100) => CSU::OutOfDetent,
+            (0b01000, 0b11000) => CSU::OutOfDetent,
+            (0b01100, 0b01000) => CSU::OutOfDetent,
+            (0b01100, 0b01100) => CSU::Conf1,
+            (0b01100, 0b00100) => CSU::OutOfDetent,
+            (0b00100, 0b00100) => CSU::OutOfDetent,
+            (0b00100, 0b00110) => CSU::OutOfDetent,
+            (0b00100, 0b01100) => CSU::OutOfDetent,
+            (0b00110, 0b00100) => CSU::OutOfDetent,
+            (0b00110, 0b00110) => CSU::Conf2,
+            (0b00110, 0b00010) => CSU::OutOfDetent,
+            (0b00010, 0b00010) => CSU::OutOfDetent,
+            (0b00010, 0b00011) => CSU::OutOfDetent,
+            (0b00010, 0b00110) => CSU::OutOfDetent,
+            (0b00011, 0b00010) => CSU::OutOfDetent,
+            (0b00011, 0b00011) => CSU::Conf3,
+            (0b00011, 0b00001) => CSU::OutOfDetent,
+            (0b00001, 0b00001) => CSU::OutOfDetent,
+            (0b00001, 0b10001) => CSU::OutOfDetent,
+            (0b10001, 0b00001) => CSU::OutOfDetent,
+            (0b00001, 0b00011) => CSU::OutOfDetent,
+            (0b10001, 0b10001) => CSU::ConfFull,
+            _ => CSU::Fault,
+        }
+    }
+}
+
+pub struct CSUMonitor {
+    handle_position_id: VariableIdentifier,
+
+    current_switch_pattern: (SwitchPattern, SwitchPattern),
+    current_detent: CSU,
+    previous_switch_pattern: (SwitchPattern, SwitchPattern),
+    previous_detent: CSU,
+
+    last_valid_switch_pattern: (SwitchPattern, SwitchPattern),
+    time_since_last_valid_detent: Duration,
+}
+impl CSUMonitor {
+    pub fn new(context: &mut InitContext) -> Self {
+        Self {
+            handle_position_id: context.get_identifier("FLAPS_HANDLE_INDEX".to_owned()),
+            current_switch_pattern: (0b11000, 0b11000),
+            current_detent: CSU::Conf0,
+            previous_switch_pattern: (0b11000, 0b11000),
+            previous_detent: CSU::Conf0,
+            last_valid_switch_pattern: (0b11000, 0b11000),
```

Comment:
Maybe these should be initialized depending on the starting state which you can get with `context.start_state()`.
This would also solve the problem with the `CSU:Misadjust` state.